githooks/prepush

```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+make pre-push
+
+status=$?
+
+# If the make command failed, prevent the push
+if [ $status -ne 0 ]; then
+  echo "Pre-push hook: make pre-push failed, aborting push."
```

Comment:
```suggestion
  echo "ERROR: Please run 'make pre-push', then 'git add' any changes, and then 'git push' a new commit."
```

Reply:
sure, though it gets copied into .git/hooks to be a pre-commit hook, and those usually are not extensioned 

The commit message, def